app/controllers/qa/issues_controller.rb

```diff
@@ -20,11 +20,19 @@ def edit
   end
 
   def update
-    @issues = current_project.issues.ready_for_review.where(id: params[:ids])
+    @issues = current_project.issues.where(id: params[:ids])
 
     respond_to do |format|
       if @issues.update_all(state: @state, updated_at: Time.now)
-        format.html { redirect_to project_qa_issues_path(current_project), notice: 'State updated successfully.' }
+        format.html do
+          # prevent attempting to redirect the user back to the show view when the QA widget is used.
+          # Record will not exist as it'll no longer be ready for review.
+          if session[:return_to] == project_qa_issue_url(current_project, params[:ids])
+            session[:return_to] = nil
+          end
```

Comment:
This is a a smell and it seems like it doesn't belong to this action. Where does it belong to?

Reply:
I split this up into `update` and `bulk_update`. Now we have removed this completely